### PR TITLE
Add prefix matching on query parameter

### PR DESCRIPTION
**Pull request by SpecialYang**
Fix https://github.com/istio/istio/issues/43710.

cc @hzxuzhonghu

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]

**Review by hzxuzhonghu — COMMENTED** [review]
I think you should also update the api comments to remove no prefix limitation

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -895,29 +895,25 @@ func translateQueryParamMatch(name string, in *networking.StringMatch) *route.Qu
 		Name: name,
 	}
 
-	switch m := in.MatchType.(type) {
-	case *networking.StringMatch_Exact:
-		out.QueryParameterMatchSpecifier = &route.QueryParameterMatcher_StringMatch{
-			StringMatch: &matcher.StringMatcher{MatchPattern: &matcher.StringMatcher_Exact{Exact: m.Exact}},
+	if isCatchAllStringMatch(in) {
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:898]: Do we have a test case for this change?

Comment (SpecialYang) [reply pilot/pkg/networking/core/v1alpha3/route/route.go:898]: Yes.

File: pilot/pkg/networking/core/v1alpha3/route/route_test.go [hunk 2]
```diff
@@ -362,6 +362,58 @@ func TestBuildHTTPRoutes(t *testing.T) {
...
+
+		routes, err := route.BuildHTTPRoutesForVirtualService(node(cg), virtualServiceWithRegexMatchingOnQueryParameter,
+			serviceRegistry, nil, 8080, gatewayNames, false, nil)
+		xdstest.ValidateRoutes(t, routes)
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+		g.Expect(len(routes)).To(gomega.Equal(1))
+		g.Expect(routes[0].GetMatch().GetQueryParameters()[0].GetStringMatch().GetSafeRegex().GetRegex()).To(gomega.Equal("BAR .+?\\..+?\\..+?"))
+	})
+
+	t.Run("for virtual service with regex matching for all cases on query parameter", func(t *testing.T) {
```
Comment (SpecialYang) [inline pilot/pkg/networking/core/v1alpha3/route/route_test.go:401]: @ramaraochavali Here is the test case for regex * case.

**Review by ericvn — APPROVED** [review]
Thanks. Release notes and test look good.

**Comment by SpecialYang** [comment]
> I think you should also update the api comments to remove no prefix limitation

Done. https://github.com/istio/api/pull/2712

**Pull request merged**